### PR TITLE
fix unstable static comment 

### PR DESCRIPTION
**Pull request by lemon79**
issue #1360 
also added tests

### CODE REVIEW

File: src/comments.js [hunk 1]
```diff
@@ -6,10 +6,12 @@ var n = types.namedTypes;
 var isArray = types.builtInTypes.array;
 var isObject = types.builtInTypes.object;
 var docBuilders = require("./doc-builders");
+var fromString = docBuilders.fromString;
 var concat = docBuilders.concat;
 var hardline = docBuilders.hardline;
 var breakParent = docBuilders.breakParent;
 var indent = docBuilders.indent;
+var align = docBuilders.align;
```
Comment (vjeux) [inline src/comments.js]: You don't need

File: src/comments.js [hunk 2]
```diff
@@ -6,10 +6,12 @@ var n = types.namedTypes;
 var isArray = types.builtInTypes.array;
 var isObject = types.builtInTypes.object;
 var docBuilders = require("./doc-builders");
+var fromString = docBuilders.fromString;
```
Comment (vjeux) [inline src/comments.js]: You don't need

File: src/comments.js [hunk 3]
```diff
@@ -143,7 +145,7 @@ function attach(comments, ast, text) {
     if (util.hasNewline(text, locStart(comment), { backwards: true })) {
       // If a comment exists on its own line, prefer a leading comment.
       // We also need to check if it's the first line of the file.
-      if (
+      if (
```
Comment (vjeux) [inline src/comments.js]: Can you set your code editor to remove trailing whitespace?

File: src/comments.js [hunk 4]
```diff
@@ -176,7 +178,8 @@ function attach(comments, ast, text) {
           precedingNode,
           comment
         ) ||
-        handleAssignmentPatternComments(enclosingNode, comment)
+        handleAssignmentPatternComments(enclosingNode, comment) ||
+		handleStaticExpressionComments(enclosingNode, followingNode, comment)
```
Comment (vjeux) [inline src/comments.js]: Use 2 spaces instead of tabs please

File: src/comments.js [hunk 5]
```diff
@@ -192,6 +195,7 @@ function attach(comments, ast, text) {
       }
     } else if (util.hasNewline(text, locEnd(comment))) {
       if (
+
```
Comment (vjeux) [inline src/comments.js]: Remove

File: src/comments.js [hunk 6]
```diff
@@ -822,7 +840,7 @@ function printLeadingComment(commentPath, print, options) {
   return concat([contents, hardline]);
 }
 
-function printTrailingComment(commentPath, print, options) {
+function printTrailingComment(commentPath, print, options, parentNode) {
```
Comment (vjeux) [inline src/comments.js]: you don't use parentNode

File: src/comments.js [hunk 7]
```diff
@@ -861,6 +879,7 @@ function printTrailingComment(commentPath, print, options) {
 }
 
 function printDanglingComments(path, options, sameIndent) {
+  const text = options.originalText;
```
Comment (vjeux) [inline src/comments.js]: you don't use this.

File: src/comments.js [hunk 8]
```diff
@@ -905,7 +924,7 @@ function printComments(path, print, options, needsSemi) {
 
     if (leading) {
       leadingParts.push(printLeadingComment(commentPath, print, options));
-
+
```
Comment (vjeux) [inline src/comments.js]: remove

File: src/comments.js [hunk 9]
```diff
@@ -808,6 +825,7 @@ function getQuasiRange(expr) {
 function printLeadingComment(commentPath, print, options) {
   const comment = commentPath.getValue();
   const contents = printComment(commentPath);
+  const text = options.originalText;
```
Comment (vjeux) [inline src/comments.js]: You don't use this

File: src/comments.js [hunk 10]
```diff
@@ -462,6 +467,18 @@ function handleMemberExpressionComments(enclosingNode, followingNode, comment) {
   return false;
 }
 
+function handleStaticExpressionComments(enclosingNode, followingNode, comment) {
+  if (
+    enclosingNode &&
+    enclosingNode.static === true
```
Comment (vjeux) [inline src/comments.js]: Outside of small things about your PR, I unfortunately don't think that this is a proper solution. We don't want to force all the comments to be leading if it is static. We only want that if the comment is between `static` and `function` keywords. I'm not really sure how we would do that with the current infrastructure to be honest.

**Comment by lemon79** [comment]
@vjeux hey i was still learning git commands, they finally started responding

**Pull request closed without merging**